scrapy_autoextract/providers.py

```diff
@@ -45,17 +51,88 @@ def __init__(
         self.stats = stats
         self.settings = settings
 
+    def __before__(self):
+        self.providers.append(self)
+
     async def __call__(self):
         """Make an AutoExtract request and build a Page Input of provided class
         based on API response data.
         """
-        page_type = self.get_page_type()
-        self.stats.inc_value(f"autoextract/{page_type}/total")
+        try:
+            if self.providers[-1] is self:
+                await self._fetch_data()
+        except IndexError:
+            # FIXME: it's not possible to use this alone yet
+            raise Exception("AutoExtractHTMLData should not be used alone")
+
+        data = await self._await_data()
+
+        if "error" in data:
+            self.stats.inc_value("autoextract/error/query")
```

Comment:
A change in code's behaviour proposed by this pull request is that now we can increment `autoextract/success` and `autoextract/error/query` at the same time. It makes sense to me, but I think it's important to highlight this change and ask for your opinions.